librz/analysis/esil/esil_trace.c

```diff
@@ -18,6 +18,84 @@ static void htup_vector_free(HtUPKv *kv) {
 	rz_vector_free(kv->value);
 }
 
+static void htpp_string_free(HtPPKv *kv) {
+	free(kv->key);
+	free(kv->value);
+}
+
+RZ_API void ht_db_array_add(HtPP *db, const char *key, const char *val) {
```

Comment:
I think it's better to move those to the `librz/util/` instead.

Reply:
Also, for new RZ_API functions please provide doxygen style doc (and unit tests, if possible).
And... why do we want this? This seems to mimic old sdb behaviour which was really bad. If you need an array, instead of just concatenating string use an HtPP <char *, RzVector *>. Using strings everywhere is not good.